packages/design-system/src/components/tree/horizontal-shift.ts

```diff
@@ -95,18 +82,24 @@ export const useHorizontalShift = <Data extends { id: string }>({
         return indexCorrected === children.length;
       };
 
-      let potentialNewParent = dropTargetPath[shifted + 1];
-
-      while (
-        isAtTheBottom(newParent, newPosition) &&
-        typeof potentialNewParent === "object" &&
-        canAcceptChild(potentialNewParent.item) &&
-        shifted < currentDepth - desiredDepth
-      ) {
-        shifted++;
-        newPosition = dropTargetPath[shifted - 1].position + 1;
-        newParent = potentialNewParent.item;
-        potentialNewParent = dropTargetPath[shifted + 1];
+      // skip drop item
+      for (let index = 1; index < dropTargetPath.length; index += 1) {
+        const potentialNewParent = dropTargetPath[index];
+        if (
+          isAtTheBottom(newParent, newPosition) &&
+          canAcceptChild(potentialNewParent) &&
+          shifted < currentDepth - desiredDepth
+        ) {
+          shifted = index;
+          newParent = potentialNewParent;
+          const child = dropTargetPath[index - 1];
+          const childPosition = getItemChildren(newParent).findIndex(
+            (item) => item.id === child.id
+          );
+          newPosition = childPosition + 1;
+          continue;
+        }
+        break;
```

Comment:
Might be subjective but `while` is easier to understand for me, at least it's less code. No strong opinion

Reply:
It's just was not very clear how path was iterated. Harder to rewrite with item selector